pkg/builder/action_merge_meta_data_builder.go

```diff
@@ -187,3 +187,120 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
 	mergeSpec.ExecutionTargets = executionTargetList
 	return mergeSpec, nil
 }
+
+
+
+
+// Horizontal Scale Merge Policy DTO builder
+type HorizontalScaleMergePolicyBuilder struct {
+	entityType                *proto.EntityDTO_EntityType
+	aggregationTargets        []*ActionAggregationTargetBuilder
+	chainedAggregationTargets []*ActionDeDuplicateAndAggregationTargetBuilder
+	commTypes                 []*CommodityMergeData
+	entityFilters              []*proto.EntityDTO_WorkloadControllerData
+}
+
+func NewHorizontalScaleMergeSpecBuilder() *HorizontalScaleMergePolicyBuilder {
+	return &HorizontalScaleMergePolicyBuilder{}
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) ForEntityType(entityType proto.EntityDTO_EntityType) *HorizontalScaleMergePolicyBuilder {
+	hsb.entityType = &entityType
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) AggregateBy(mergeTarget *ActionAggregationTargetBuilder) *HorizontalScaleMergePolicyBuilder {
+	hsb.aggregationTargets = append(hsb.aggregationTargets, mergeTarget)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) DeDuplicateAndAggregateBy(mergeTarget *ActionDeDuplicateAndAggregationTargetBuilder) *HorizontalScaleMergePolicyBuilder {
+	hsb.chainedAggregationTargets = append(hsb.chainedAggregationTargets, mergeTarget)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) ForWorkloadControllerDataFilter(controllerData proto.EntityDTO_WorkloadControllerData) *HorizontalScaleMergePolicyBuilder  {
```

Comment:
Shouldn't this be a `ContainerPodData` object? I thought we originally had `WorkloadControllerData` in the proto, but changed it to `ContainerPodData` after our discussions.

Reply:
Yes. Thanks!